### DISCUSSION TITLE
How about support posting a delay task in App::Post() or support a timer event

### DISCUSSION THREAD

**Discussion by OUHHDFF** [ArthurSonzogni/FTXUI#1257] [Ideas]
If so, it will be convient to write something like this:
```
    std::string label { "hello" };
    auto btn = Button(&label, [&] () -> Helper::Task { 
        co_await 1s;
        label = "1s later";
    });
    app.Loop(btn);
```
because you can write :

```
void await_suspend(std::coroutine_handle<> h) noexcept { 
           app.Post(1s, [=]{ h.resume(); });
 }    
```
and it doesn't require anothre thread, that's cool.

**Comment by ArthurSonzogni** [comment]
I am supportive!

I recently:
- Removed usage of threads.
- Added a proper internal task scheduler, supporting delays.

This should gives the proper tools to implement what you ask. This is what I originally envisioned, but haven't yet implemented it.

With a new baby, I am lacking time to add new features. Happy to do it myself, but if you have some free time next to you, I am also fine receiving pull requests!

**Reply by OUHHDFF** [reply]
Thank you for your reply. Frankly speaking, I haven't code for a long time and just picked it up last serval month. Therefore I'm not confident in submitting code.
Besides, this is a rough idea, and I haven't verified concept behind so far. If there is any intention to support it, perhaps more consideration needs to be given.
Finally, congratulations on your new baby !

**Comment by OUHHDFF** [comment]
I find it can be done by a custom loop and maintain a time point heap 🙄

**Comment by OUHHDFF** [comment]
I attempted to implement my initial idea and discovered that the biggest issue is that although std::function<void()> now supports coroutines, the lifecycle of the coroutine is also transferred to std::function. This results in the destruction of the function object to which the coroutine function belongs after the first co_await, causing the program to crash when the coroutine resumes. std::function_view can help solve this problem, but it will have to wait until C++26 is implemented, now its like this:

```
    std::string label { "hello" };
    auto g = [&] () -> Helper::Task { 
        co_await 1s;
        label = "1s later";
     })；
    auto btn = ftxui::Button(&showText2, [&g] { g().detach(); });
```
